sqlalchemy-stubs/sql/sqltypes.pyi

```diff
@@ -243,7 +244,9 @@ class Interval(Emulated, _AbstractInterval[timedelta], TypeDecorator[timedelta])
     def bind_processor(self, dialect: Dialect) -> Callable[[Optional[timedelta]], str]: ...
     def result_processor(self, dialect: Dialect, coltype: Any) -> Callable[[Optional[str]], Optional[timedelta]]: ...
 
-class JSON(Indexable, TypeEngine[Dict[str, Any]]):
+_JSONT = Union[Dict[str, Any], List[Any]]
```

Comment:
Did this cause some problems in actual code? If this is made from the point of view of general correctness, then I wouldn't do this.

Even if it caused some problems, I think this will likely cause lots of false positives, reading this attribute from a model will need to be always followed by an `isinstance(..., dict)`.

Reply:
We have multiple JSON fields that look like `[{"from": 1, "to": "3", "name": "x"}, ...]`. So in our case it created a lot of false negatives.
In PostgreSQL (and hence SQLAlchemy) there as many operators to handle `[{"from": 1, "to": "3", "name": "x"}, ...]` as there are for `{"from": 1, "to": "3", "name": "x"}`. So I thinks It's a very common practice to have lists. https://www.postgresql.org/docs/9.3/functions-json.html 

Maybe a solution would be to give a hint when creating the column ?

Reply:
Sorry for a delay, coming back to this.

> So in our case it created a lot of false negatives.

I would say that false negatives are less important that false positives, the other changes in this PR are OK however.

Reply:
Sorry, I meant false positives. Again, a JSON field is not a dict. It is very common to store smth like `[{"from": 1, "to": "3", "name": "x"}, ...]`. It is both supported by Postgres and SQLAlchemy.
This causes any code such as `MyModel.my_json_col[0]` to be marked as an error. Adding a type ignore to every occurence is not an option IMO.
What is the rational behind setting JSONB to be a dict ?